In [1]:
import json
import numpy as np
from sklearn.ensemble import RandomForestRegressor
from sklearn import metrics
from sklearn.model_selection import train_test_split

In [8]:
from sklearn.model_selection import GridSearchCV

In [2]:
file = open('dataset/vectorized_spacy.json','r')
data = json.load(file)
x_data = []
y_data = []
for i in data.keys():
    x_data.append(data[i][2])
    y_data.append(data[i][1])
x_train, x_test, y_train, y_test = train_test_split(x_data, y_data, test_size=0.20)

In [3]:
regressor = RandomForestRegressor(n_estimators=20, random_state= 0)
regressor.fit(x_train,y_train)

RandomForestRegressor(n_estimators=20, random_state=0)

### Tried different combinations

spending 30 minutes is also not worth it

In [17]:
regressor = RandomForestRegressor()
param_grid = {
   'n_estimators': [100, 200, 500],
   # 'max_features': ['auto', 'sqrt', 'log2'],
   'max_depth' : [4,5,6,8,9],
   'criterion' :['squared_error']
}
GSCV = GridSearchCV(estimator=regressor, param_grid=param_grid, cv=10)
model = GSCV.fit(x_train, y_train)
# GSCV.best_params_ 

In [18]:
pred = model.predict(x_test)

In [19]:
mse = metrics.mean_absolute_error(y_test,pred)
print(mse)

0.16852292871487562


#### original result

In [4]:
y_pred = regressor.predict(x_test)

In [5]:
mse = metrics.mean_absolute_error(y_test,y_pred)
print(mse)

0.16766116884989113


In [7]:
r2_score = metrics.r2_score(y_test, y_pred)
print(r2_score)

0.16668782139952787


If n_estimators are more then mse will be less but it will take more time. for eg if n_estimator = 200 mse will be around 0.169 